mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -351,17 +359,21 @@ private[ml] object DefaultParamsReader {
       timestamp: Long,
       sparkVersion: String,
       params: JValue,
+      defaultParams: JValue,
       metadata: JValue,
       metadataJson: String) {
 
     /**
      * Get the JSON value of the [[org.apache.spark.ml.param.Param]] of the given name.
      * This can be useful for getting a Param value before an instance of `Params`
      * is available.
+     *
+     * @param isDefaultParam Whether the given param name is a default param. Default is false.
      */
-    def getParamValue(paramName: String): JValue = {
+    def getParamValue(paramName: String, isDefaultParam: Boolean = false): JValue = {
```

Comment:
Should the logic be "lookup params first, if not exist then lookup defaultParams" ?
If so, we don't need the `isDefaultParam` param.